### PR TITLE
[DEV-1481] Create collection

### PR DESCRIPTION
**Pull request by mihaiberq**
Base code to allow for `RefinableWhitelistedToken` contract creation in the sdk

### CODE REVIEW

File: src/refinable/Contracts.ts [hunk 1]
```diff
@@ -186,4 +194,46 @@ export class Contracts {
...
+      metadataUri,
+      ipfsUri, // uri
+      ...optionalParam(is1155, ipfsUri)
+    );
+
+    await contract.deployed();
+
+    await contract.addMinter(this.refinable.accountAddress);
+
+    return new EvmTransaction(contract.deployTransaction);
```
Comment (sneljo1) [inline src/refinable/Contracts.ts:237]: Are we able to retrieve the contract address from this? Otherwise would make sure that we do

**Review by sneljo1 — COMMENTED** [review]
I think we should either create the contract via our API after deployment, or create a separate method to do so. Because right now this is a bit useless. If the user uses the SDK standalone. yes he created his contract, but can't even use it with us.

File: src/refinable/Contracts.ts (hunk 1 above)
Comment (mihaiberq) [reply src/refinable/Contracts.ts:237]: yes, im also displaying contract address + tx hash in the modal + success page.

**Comment by mihaiberq** [comment]
hmmm ok, ill add a separate function to create the contract

**Comment by sneljo1** [comment]
> hmmm ok, ill add a separate function to create the contract

If we do that, we'll need to make it clear tho to the user that this is required

File: src/refinable/Contracts.ts [hunk 2]
```diff
@@ -186,4 +199,75 @@ export class Contracts {
...
+    const { createContract: dbContractResponse } =
+      await this.refinable.apiClient.request<
+        CreateContractMutation,
+        CreateContractMutationVariables
+      >(CREATE_CONTRACT, {
+        data: {
+          contract: {
+            contractAddress: tx.txReceipt.contractAddress,
+            chainId: await this.refinable.provider.getChainId(),
+            contractType: type,
```
Comment (sneljo1) [inline src/refinable/Contracts.ts:260]: I would send the abi, less work for the backend.

File: src/refinable/Contracts.ts [hunk 3]
```diff
@@ -186,4 +199,75 @@ export class Contracts {
...
     return contract;
   }
+
+  async createContract(
+    type:
+      | ContractTypes.Erc721WhitelistedToken
+      | ContractTypes.Erc1155WhitelistedToken,
+    name: string,
+    ticker: string,
+    collection: SdkCollectionInput
```
Comment (sneljo1) [inline src/refinable/Contracts.ts:209]: All of these different params are a bit messy. Would make more sense if the contract properties are a property of the collection input as collection is the parent?

**Review by sneljo1 — CHANGES_REQUESTED** [review]

File: src/refinable/Contracts.ts (hunk 2 above)
Comment (mihaiberq) [reply src/refinable/Contracts.ts:260]: i would still have to verify the abi in the backend (not to allow random, broken abis to be sent to that endpoint), so id rather keep it like this

**Review by sneljo1 — APPROVED** [review]
LGTM

**Pull request merged**